src/test/java/org/minions/devfund/dennis/SquarelotronTest.java

```diff
@@ -0,0 +1,202 @@
+     */
+    @Test
+    public void testRotateRight2() {
+        Squarelotron se = new Squarelotron(TWO);
+        se.rotateRight(-1);
+        assertEquals(TWO, se.getMatrix()[ZERO][ZERO]);
+        assertEquals(FOUR, se.getMatrix()[ZERO][ONE]);
+        assertEquals(ONE, se.getMatrix()[ONE][ZERO]);
+        assertEquals(THREE, se.getMatrix()[ONE][ONE]);
+    }
+
+    /**
+     * Unit test for testRotateRight where n = 2 and numberOfTurns = 2.
+     */
+    @Test
+    public void testRotateRight3() {
+        Squarelotron se = new Squarelotron(TWO);
+        se.rotateRight(TWO);
+        assertEquals(FOUR, se.getMatrix()[ZERO][ZERO]);
+        assertEquals(THREE, se.getMatrix()[ZERO][ONE]);
+        assertEquals(TWO, se.getMatrix()[ONE][ZERO]);
+        assertEquals(ONE, se.getMatrix()[ONE][ONE]);
+    }
+
+    /**
+     * Unit test for testRotateRight where n = 3 and numberOfTurns = 1.
+     */
+    @Test
+    public void testRotateRight4() {
+        Squarelotron se = new Squarelotron(THREE);
+        se.rotateRight(ONE);
+        assertEquals(ONE, se.getMatrix()[ZERO][TWO]);
+        assertEquals(TWO, se.getMatrix()[ONE][TWO]);
+        assertEquals(THREE, se.getMatrix()[TWO][TWO]);
+        assertEquals(FOUR, se.getMatrix()[ZERO][ONE]);
+        assertEquals(FIVE, se.getMatrix()[ONE][ONE]);
+    }
+
+    /**
+     * Unit test for numbersInRing where n = 4 true.
+     */
+    @Test
+    public void testIsOnRing() {
+        Squarelotron se = new Squarelotron(FOUR);
+        assertTrue(se.numbersInRing(ZERO, ZERO, ONE));
+        assertTrue(se.numbersInRing(ONE, ZERO, ONE));
+        assertTrue(se.numbersInRing(THREE, ZERO, ONE));
+        assertTrue(se.numbersInRing(ZERO, THREE, ONE));
+        assertTrue(se.numbersInRing(ONE, ONE, TWO));
+        assertTrue(se.numbersInRing(TWO, TWO, TWO));
+    }
+
+    /**
+     * Unit test for numbersInRing where n = 4 false.
+     */
+    @Test
+    public void testIsOnRing2() {
+        Squarelotron se = new Squarelotron(FOUR);
+        assertFalse(se.numbersInRing(ZERO, ZERO, TWO));
+        assertFalse(se.numbersInRing(ZERO, ONE, TWO));
+        assertFalse(se.numbersInRing(THREE, ZERO, TWO));
+        assertFalse(se.numbersInRing(ONE, ONE, ONE));
+        assertFalse(se.numbersInRing(ONE, TWO, ONE));
+    }
+
+    /**
+     * Unit test for UpsideDownFlip where n = 4 ring = 1.
+     */
+    @Test
+    public void testUpsideDownFlip1() {
+        Squarelotron se = new Squarelotron(FOUR);
+        se = se.upsideDownFlip(ONE);
+        assertEquals(THIRTEEN, se.getMatrix()[ZERO][ZERO]);
+        assertEquals(FIFTEEN, se.getMatrix()[ZERO][TWO]);
+        assertEquals(NINE, se.getMatrix()[ONE][ZERO]);
+        assertEquals(SIX, se.getMatrix()[ONE][ONE]);
+        assertEquals(EIGHT, se.getMatrix()[TWO][THREE]);
+        assertEquals(FOUR, se.getMatrix()[THREE][THREE]);
+    }
+
+    /**
+     * Unit test for UpsideDownFlip where n = 4 ring = 2.
+     */
+    @Test
+    public void testUpsideDownFlip2() {
```

Comment:
In order to make understandable your unit tests
Try to re-writing something like this.
```java
    @Test
    public void testUpsideDownFlipDimension4Ring2() {
        final int ringNumber = 2;
        final int dimension = 4;
        final Squarelotron squarelotron = new Squarelotron(dimension);
        final int[][] expectedMatrix = {{1, 2, 3, 4}, {5, 10, 11, 8}, {9, 6, 7, 12}, {13, 14, 15, 16}};
        Squarelotron result = squarelotron.upsideDownFlip(ringNumber);
        assertArrayEquals(expectedMatrix, result.getMatrix());
    }
```